libdnf5/repo/repo.cpp

```diff
@@ -520,38 +520,67 @@ void Repo::internalize() {
-        repo_cachedir, base_cachedir);
+    libdnf_assert(
+        base_path_pos != std::string::npos,
+        "Repo cachedir \"{}\" doesn't contain base cachedir \"{}\"",
+        repo_cachedir,
+        base_cachedir);
 
     auto root_repo_cachedir = repo_cachedir;
-    root_repo_cachedir.replace(base_path_pos, base_cachedir.size(),
-        config.get_main_config().get_system_cachedir_option().get_value());
+    root_repo_cachedir.replace(
+        base_path_pos, base_cachedir.size(), config.get_main_config().get_system_cachedir_option().get_value());
 
     auto root_repodata_cachedir = std::filesystem::path(root_repo_cachedir) / CACHE_METADATA_DIR;
-    if (std::filesystem::exists(root_repodata_cachedir)) {
-        auto repo_cache = RepoCache(base, repo_cachedir);
-        repo_cache.remove_metadata();
-        repo_cache.remove_solv_files();
+    try {
+        if (!std::filesystem::exists(root_repodata_cachedir)) {
+            return false;
+        }
+    } catch (const std::filesystem::filesystem_error & e) {
+        logger.debug("Error when checking root repodata at \"{}\" : \"{}\"", root_repodata_cachedir.c_str(), e.what());
+        return false;
+    }
 
-        auto repodata_cachedir = std::filesystem::path(repo_cachedir) / CACHE_METADATA_DIR;
+    auto repo_cache = RepoCache(base, repo_cachedir);
+    repo_cache.remove_metadata();
+    repo_cache.remove_solv_files();
+
+    auto repodata_cachedir = std::filesystem::path(repo_cachedir) / CACHE_METADATA_DIR;
+    try {
         std::filesystem::create_directories(repodata_cachedir);
         std::filesystem::copy(root_repodata_cachedir, repodata_cachedir);
+    } catch (const std::filesystem::filesystem_error & e) {
+        logger.debug(
+            "Error when cloning root repodata from \"{}\" to \"{}\" : \"{}\"",
+            root_repodata_cachedir.c_str(),
+            repodata_cachedir.c_str(),
+            e.what());
+        repo_cache.remove_metadata();
+        return false;
+    }
 
-        auto root_solv_cachedir = std::filesystem::path(root_repo_cachedir) / CACHE_SOLV_FILES_DIR;
-        auto solv_cachedir = std::filesystem::path(repo_cachedir) / CACHE_SOLV_FILES_DIR;
+    auto root_solv_cachedir = std::filesystem::path(root_repo_cachedir) / CACHE_SOLV_FILES_DIR;
+    auto solv_cachedir = std::filesystem::path(repo_cachedir) / CACHE_SOLV_FILES_DIR;
+    try {
         if (std::filesystem::exists(root_solv_cachedir)) {
             std::filesystem::create_directories(solv_cachedir);
             std::filesystem::copy(root_solv_cachedir, solv_cachedir);
         }
-
-        return true;
-    } else {
-        return false;
+    } catch (const std::filesystem::filesystem_error & e) {
+        logger.debug(
+            "Error when cloning root solv data from \"{}\" to \"{}\" : \"{}\"",
+            root_solv_cachedir.c_str(),
+            solv_cachedir.c_str(),
+            e.what());
+        repo_cache.remove_solv_files();
     }
+
+    return true;
```

Comment:
I would prefer to also squash this commit but this might be a question of taste.

Reply:
Sure I can, this is just a result of implementing it incrementally.